File: LeetCode Problems/301 to 400/374. Guess Number Higher or Lower/374. Guess Number Higher or Lower.cpp
/**
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */
typedef long long int ll;
class Solution {
public:
    int guessNumber(int n) {
        ll f = 1;
        while (f <= n) {
            int m = (f + n) / 2;
            int gues = guess(m);
            if (gues == 0) {
                return m;
            } else if (gues == 1) {
                f = m + 1;
            } else
                n = m - 1;
        }
        return 0;
    }
};

File: LeetCode Problems/301 to 400/374. Guess Number Higher or Lower/374. Guess Number Higher or Lower.java
/**
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int l = 1, m, pick;
        while (l <= n) {
            m = l + (n - l) / 2;
            pick = guess(m);
            if (pick == 0)
                return m;
            else if (pick == -1)
                n = m - 1;
            else
                l = m + 1;
        }
        return 0;
    }
}
